content/05.dl-complexities.md

```diff
@@ -2,13 +2,13 @@
 
 Correctly training deep neural networks is a non-trivial process.
 There are many different options and potential pitfalls at every stage.
-To get good results you have to expect to train many networks with a range of different parameter and hyperparameter settings.
+To get good results, you must expect to train many networks with a range of different parameter and hyperparameter settings.
 Despite improved framework ease-of-use and on-demand cloud computing resources this means DL can be very demanding, often requiring significant infrastructure and patience to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740].
```

Comment:
This line doesn't seem to correct, but I'm not quite sure what framework ease-of-use and DL infrastructure requirements have in common. Can we make this line clearer?

Reply:
To make this syntactically (but not necessarily semantically) correct, how about:

"Despite improved framework ease-of-use and on-demand cloud computing resources, DL can be very demanding, often requiring significant infrastructure and patience to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740]."

Reply:
Should we start the sentence with "DL can be very demanding" and move the initial clause elsewhere?  I think the original idea was that even though it is easier to implement and run DL models, that doesn't change the local or cloud computational demands.

Reply:
How about:
> DL [training?] can be very demanding, often requiring significant computing infrastructure and patience to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740].

Reply:
Yes, I like that.  I lean toward excluding "training" in case that implies training a single model is hard.  Often the real demand is in hyperparameter search and training many models.

Reply:
What about "model optimization" or "hyperparameter search" instead of training?

Reply:
You're right; there's no need for that word since the last sentence makes it clear what we're talking about